Tests/NIOTransportServicesTests/NIOTSConnectionChannelTests.swift

```diff
@@ -865,5 +865,61 @@ class NIOTSConnectionChannelTests: XCTestCase {
             .wait()
         XCTAssertNoThrow(try connection.close().wait())
     }
+
+    func testErrorIsForwardedFromFailedConnectionState() throws {
+        final class ForwardErrorHandler: ChannelDuplexHandler {
+            typealias OutboundIn = ByteBuffer
+            typealias InboundIn = ByteBuffer
+            
+            private let testCompletePromise: EventLoopPromise<Error>
+            let listenerChannel: Channel
+            
+            init(testCompletePromise: EventLoopPromise<Error>, listenerChannel: Channel) {
+                self.testCompletePromise = testCompletePromise
+                self.listenerChannel = listenerChannel
+            }
+            func channelActive(context: ChannelHandlerContext) {
+                let listenerErrorPromise = self.listenerChannel.eventLoop.next().makePromise(of: Error.self)
+                listenerChannel
+                    .close()
+                    .whenSuccess { _ in
+                        context.channel.write(ByteBuffer(data: Data()))
+                            .whenFailure({ error in
+                                listenerErrorPromise.succeed(error)
+                            })
```

Comment:
I don't think this path should complete the promise: we're testing for this happening in errorcaught, not on the write promise.

Reply:
Are you cool with me just throwing the result of the write away?